### PR TITLE
if coredump is to big, delete it instead of archiving

### PR DESCRIPTION
**Pull request by dothebart**

### CODE REVIEW

**Comment by dothebart** [comment]
https://jenkins01.arangodb.biz/job/arangodb-ANY-linux-san.x86-64/1330/ including other changes.

File: jenkins/helper/test_launch_controller.py [hunk 1]
```diff
@@ -111,6 +112,22 @@ def get_workspace():
...
 os.environ['TMP'] = str(TEMP)
 
+MAX_COREFILES_SINGLE=4
+MAX_COREFILES_CLUSTER=15
+if 'MAX_CORECOUNT' in os.environ:
+    MAX_COREFILES_SINGLE=int(os.environ['MAX_CORECOUNT'])
+    MAX_COREFILES_CLUSTER=int(os.environ['MAX_CORECOUNT'])
+MAX_TOTAL_CORESIZE_MB=1500
+if 'MAX_TOTAL_CORESIZE' in os.environ:
+    MAX_TOTAL_CORESIZE_MB=int(os.environ['MAX_TOTAL_CORESIZE'])
```
[the remark this answers is not in the document]
Comment (dothebart) [reply jenkins/helper/test_launch_controller.py:122]: I think yes, we need this. You want to archive no more than `n` corefiles, up to `m` bytes individual size, `o` bytes combined size.
even if `n` goes down to a size way below `o` in result.

File: jenkins/helper/test_launch_controller.py [hunk 2]
```diff
@@ -726,11 +743,40 @@ def generate_crash_report(self):
...
+                    if delete_it:
+                        size_count -= size
+                        print(f'deleting coredump {str(one_file)}')
+                        files.remove(one_file)
+                        files_unsorted.remove(one_file)
+
+        if len(files_unsorted) > core_max_count:
             count = 0
-            for one_crash_file in files:
+            for one_crash_file in files_unsorted:
```
[the remark this answers is not in the document]
Comment (dothebart) [reply jenkins/helper/test_launch_controller.py]: we have two loops here:
- if single files are to big or combined size is to big the loop you marked.
- below a loop if the remaining count is to high
But yes, should indicate why its been deleted.

File: jenkins/helper/test_launch_controller.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (KVS85) [reply jenkins/helper/test_launch_controller.py:122]: Yes. That's why we don't need it as a separate variable. Max total size is either calculated or ignored (if option regarding number of cores was off).

File: jenkins/helper/test_launch_controller.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (KVS85) [reply jenkins/helper/test_launch_controller.py]: It's not implemented in the way I described. Current implementation is non-optimal and doesn't take options to turn off either number of cores or individual core size or both.

[the remark this answers is not in the document]
Comment (dothebart) [reply jenkins/helper/test_launch_controller.py]: the msising disabling of the count condition was added:
```
        total_files_too_big = 0 < MAX_TOTAL_CORESIZE_MB < size_count
                if 0 < MAX_COREFILE_SIZE_MB < size:
                if 0 < MAX_TOTAL_CORESIZE_MB < size_count:
        if len(files_unsorted) > core_max_count > 0:
```
work down to putting `-1` to any of these conditions to disable them, hence the comparisons to `0` to implement this.
Tripple-compares seem to be a bit odd on first sight, but pylint wanted it to be done like that.

**Comment by dothebart** [comment]
https://jenkins01.arangodb.biz/view/PR/job/arangodb-matrix-pr/24997/

**Comment by dothebart** [comment]
https://jenkins01.arangodb.biz/view/PR/job/arangodb-matrix-pr/25061/ run green except for real test failures.

**Review by naushniki — APPROVED** [review]

**Review by KVS85 — CHANGES_REQUESTED** [review]
`MAX_TOTAL_CORESIZE_MB` should be removed, global loop over sorted (asc) by sizes coredumps should be used with `core_max_count` condition check and `MAX_COREFILE_SIZE_MB` break.

File: jenkins/helper/test_launch_controller.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (KVS85) [reply jenkins/helper/test_launch_controller.py]: `len(files_unsorted) > core_max_count > 0`

These lines are hard to interpret. Please use conjunction of expressions.

**Comment by dothebart** [comment]
https://jenkins01.arangodb.biz/view/PR/job/arangodb-matrix-pr/25242/

**Review by KVS85 — APPROVED** [review]

**Pull request merged**